### DISCUSSION TITLE
Allow plugins to specify `purge` patterns that are merged with the config `purge`

### DISCUSSION THREAD

**Discussion by jquense** [tailwindlabs/tailwindcss#5208] [Ideas]
👋   Hey folks, quick thought and idea. we've been experimenting with using tailwind as the styling base for installable UI libraries, for our work design system as well as OSS libraries, all on npm or github packages. These libraries are React components (tho same would be true for any component framework) and one annoyance we have is that in order for consumers to use JIT mode, they need to add the installed package to their config `purge` which is awkward and a bit leaky (I don't want users to have to know where on disk the package is installed or it's internal structure).

```js
purge: [
  './src/**/*',
  './node_modules/cool-button/lib/**/*'
]
```
The obvious thing (to me) to make this nicer would be to allow plugins to specify their own `purge` setting that would be merged in with the user config one.

```js
// cool-button/index.js

export default plugin(() => {
  /* ... */
}, {
  // ...
  purge: [`${__dirname}/**/*.js`]
})
```

Thoughts? Does this sound feasible?

**Comment by borisghidaglia** [comment]
Any news on this today?

**Reply by wongjn** [reply]
I don't think there will be – `purge` was an option in Tailwind v2 but this option no longer exists in v3+.

**Reply by melMass** [reply]
The concept is the same, the idea would be to allow extending `content` with path relative to the CL
